stack.py

```diff
@@ -0,0 +1,21 @@
+from linkedlist import LinkedList
+
+
+class Stack:
+    def __init__(self, values=None):
+        self.ll = LinkedList(values)
+        self._length = 0
+
+    def push(self, val):
+        self.ll.push(val)
+        self._length += 1
+
+    def pop(self):
+        if self.ll.head:
+            self._length -= 1
+            return self.ll.pop()
```

Comment:
Don't check if there is a head, call storage.pop and handle the exception is there is one.